### ISSUE TITLE
shellcheck disable fails for SC2103 

### ISSUE THREAD

**Issue by gnzlbg** [koalaman/shellcheck#1591]
In the following script, `# shellcheck=SC2103` fails to disable the warning, resulting in shellcheck still erroring on our CI. Reproducer: 

```sh
#!/usr/bin/env sh
#
# Install musl and musl-sanitized linux kernel headers
# to musl-{$1} directory

set -ex

MUSL_VERSION=1.1.22
MUSL="musl-${MUSL_VERSION}"

# Download, configure, build, and install musl:
curl --retry 5 https://www.musl-libc.org/releases/${MUSL}.tar.gz | tar xzf -

cd $MUSL
case ${1} in
    aarch64)
        musl_arch=aarch64
        kernel_arch=arm64
        CC=aarch64-linux-gnu-gcc \
          ./configure --prefix="/musl-${musl_arch}" --enable-wrapper=yes
        make install -j4
        ;;
    arm)
        musl_arch=arm
        kernel_arch=arm
        CC=arm-linux-gnueabihf-gcc CFLAGS="-march=armv6 -marm -mfpu=vfp" \
          ./configure --prefix="/musl-${musl_arch}" --enable-wrapper=yes
        make install -j4
        ;;
    i686)
        # cross-compile musl for i686 using the system compiler on an x86_64
        # system.
        musl_arch=i686
        kernel_arch=i386
        # Specifically pass -m32 in CFLAGS and override CC when running
        # ./configure, since otherwise the script will fail to find a compiler.
        CC=gcc CFLAGS="-m32" \
          ./configure --prefix="/musl-${musl_arch}" --disable-shared --target=i686
        # unset CROSS_COMPILE when running make; otherwise the makefile will
        # call the non-existent binary 'i686-ar'.
        make CROSS_COMPILE= install -j4
        ;;
    x86_64)
        musl_arch=x86_64
        kernel_arch=x86_64
        ./configure --prefix="/musl-${musl_arch}"
        make install -j4
        ;;
    *)
        echo "Unknown target arch: \"${1}\""
        exit 1
        ;;
esac


# shellcheck disable=SC2103
cd ..
rm -rf $MUSL

# Download, configure, build, and install musl-sanitized kernel headers:
curl --retry 5 -L \
     https://github.com/sabotage-linux/kernel-headers/archive/v4.4.2-1.tar.gz | \
    tar xzf -
(
    cd kernel-headers-4.4.2-1
    make ARCH="${kernel_arch}" prefix="/musl-${musl_arch}" install -j4
)
rm -rf kernel-headers-4.4.2-1
```

---

```
shellcheck --version
ShellCheck - shell script analysis tool
version: 0.6.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net
```

**Comment by kai-alpha** [comment]
I had the same problem and came across this ticket.

Just found out it's the first `cd $MUSL` which needs be annotated with `shellcheck disable=SC2103` for the latest shellcheck 0.6.0

**Comment by nodakai** [comment]
I had the same problem and came across this ticket.

Just found out it's the first `cd $MUSL` which needs be annotated with `shellcheck disable=SC2103` for the latest shellcheck 0.6.0

**Comment by koalaman** [comment]
This happened because the warning incorrectly tagged the annotation itself rather than the annotated command. Since annotations ignore warnings in children but not in itself, the warning was still emitted. This has now been fixed. Thanks!
